xbmc/dbwrappers/qry_dat.cpp

```diff
@@ -165,75 +170,89 @@ field_value::field_value(const field_value& fv)
   is_null = fv.get_isNull();
 }
 
+field_value::field_value(field_value&& fv) noexcept
+{
+  *this = std::move(fv);
+}
+
 //empty destructor
 field_value::~field_value() = default;
 
 //Conversations functions
-std::string field_value::get_asString() const
+std::string field_value::get_asString() const&
```

Comment:
needed a while to figure it is `const lvalue` overload and not const reference bound /-:

also would this member ever get picked?
```cpp
void DoSomething() const&& { std::cout << "const rvalue?! does such a thing exist?" << std::endl; };
```
it compiles fine

Reply:
> needed a while to figure it is const lvalue overload and not const reference bound /-:

Yeah, ref-qualifiers aren't a very well known feature of the language :)

> also would this member ever get picked?

You could force scenarios where such a member would be picked [(example)](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:25,endLineNumber:5,positionColumn:25,positionLineNumber:5,selectionStartColumn:25,selectionStartLineNumber:5,startColumn:25,startLineNumber:5),source:'%23include+%3Cstring%3E%0A%23include+%3Cfmt/core.h%3E%0A%0Astruct+S+%7B%0A++++void+print()+const%26%26+%7B+fmt::print(%22const+rval%22)%3B+%7D%0A++++void+print()+%26%26+%7B+fmt::print(%22rval%22)%3B+%7D%0A%7D%3B%0A%0Aint+main()+%7B%0A++++const+S+s%3B%0A++++std::move(s).print()%3B%0A%7D%0A%0A'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),header:(),k:53.37979094076655,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:gsnapshot,deviceViewOpen:'1',filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!((name:fmt,ver:trunk)),options:'-std%3Dc%2B%2B20+-O3',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+(trunk)+(Editor+%231)',t:'0')),header:(),k:50,l:'4',m:40.07050528789659,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+(trunk)+(Compiler+%231)',t:'0')),header:(),l:'4',m:59.929494712103406,n:'0',o:'',s:0,t:'0')),k:46.62020905923345,l:'3',n:'0',o:'',t:'0')),l:'2',m:100,n:'0',o:'',t:'0')),version:4) but I haven't seen any use for that so far. In all occasions the `const&&` method would just duplicate the `const&` version.